### ISSUE TITLE
Option to pretty-print test failures

### ISSUE THREAD

**Issue by eigenhombre** [slagyr/speclj#128]
It'd be awful nice if the `should` family could be made to pretty-print the contextual data it shows.

E.g.

    1) org-parser can handle double newlines in body
    Expected: [[:document [:body "part a\n\npart b\n"]]]
          got: ([:document [:body "part a\n\npart b\n"]] [:document [:body "part a\n"] [:body "\npart b\n"]] [:document [:body "part a\n\n"] [:body "part b\n"]] [:document [:body "part a\n"] [:body "\n"] [:body "part b\n"]] [:document [:body "part a\n"] [:body "\n"] [:body "part b\n"]] [:document [:body "part a\n"] [:body "\npart b\n"]] [:document [:body "part a\n\n"] [:body "part b\n"]] [:document [:body "part a\n\npart b\n"]]) (using =)

Would be instead:

        1) org-parser can handle double newlines in body
    Expected: [[:document [:body "part a\n\npart b\n"]]]
          got:   ([:document [:body "part a\n\npart b\n"]]
                  [:document [:body "part a\n"] [:body "\npart b\n"]]
                  [:document [:body "part a\n\n"] [:body "part b\n"]]
                  [:document [:body "part a\n"] [:body "\n"] [:body "part b\n"]]
                  [:document [:body "part a\n"] [:body "\n"] [:body "part b\n"]]
                  [:document [:body "part a\n"] [:body "\npart b\n"]]
                  [:document [:body "part a\n\n"] [:body "part b\n"]]
                  [:document [:body "part a\n\npart b\n"]])
          (using =)

**Comment by sdegutis** [comment]
Good idea! We talked about this briefly before in #45 and considered using https://github.com/brandonbloom/fipp, but that library turned out not to be a good fit for our requirements. Maybe there's another existing Clojure pretty-printer out there somewhere we can use? Otherwise, it doesn't seem hard to write a pretty-printer ourselves, so perhaps we should just write one.
